### PR TITLE
chore(typescript): upgrade react-intl

### PR DESCRIPTION
**Pull request by laughnan**
⚠️ super draft PR! still a major WIP.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @laughnan has signed the CLA. Thanks for the pull request!

File: jest.config.js [hunk 1]
```diff
@@ -4,7 +4,6 @@ module.exports = {
     clearMocks: true,
     restoreMocks: true,
     moduleNameMapper: {
-        'react-intl-locale-data': '<rootDir>/node_modules/react-intl/locale-data/en.js',
```
Comment (priyajeet) [inline jest.config.js]: You might have to add https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#jest to line 13. Also there is another jest.config.js inside scripts folder I think, double check if changes are needed there. That one is used for visual tests only.

File: scripts/webpack.config.js [hunk 2]
```diff
@@ -71,7 +70,6 @@ function getConfig(isReactExternalized) {
             alias: {
                 'box-ui-elements/es': path.join(__dirname, '../src'), // for examples only
                 examples: path.join(__dirname, '../examples/src'), // for examples only
-                'react-intl-locale-data': path.resolve(`node_modules/react-intl/locale-data/${locale}`),
```
Comment (priyajeet) [inline scripts/webpack.config.js]: https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#webpack might have to add to line 91

File: src/elements/README.md [hunk 3]
```diff
@@ -7,12 +7,3 @@ The UI Elements use [react-intl](https://github.com/yahoo/react-intl) for intern
 The `language` property is a string that can be one of `en-AU`, `en-CA`, `en-GB`, `en-US`, `bn-IN`, `da-DK`, `de-DE`, `es-419`, `es-ES`, `fi-FI`, `fr-CA`, `fr-FR`, `hi-IN`, `it-IT`, `ja-JP`, `ko-KR`, `nb-NO`, `nl-NL`, `pl-PL`, `pt-BR`, `ru-RU`, `sv-SE`, `tr-TR`, `zh-CN`, `zh-TW`.
 
 The `messages` property is a map of message keys and translated strings. All the messages that the UI elements use can be found under the [i18n](i18n) folder. We distribute them as JS modules within the `box-ui-elements` npm package and they can be imported like any other module. We provide translated strings for all the langauges listed above.
-
-For `react-intl` to work properly, locale data needs to be added via [addlocaledata](https://github.com/yahoo/react-intl/wiki/API#addlocaledata). They are all bundled inside the `react-intl` package inside `node_modules`. By default it uses the `en` locale but for any other locale you have to do similar to the following before rendering the component:
```
Comment (priyajeet) [inline src/elements/README.md]: Maybe replace this section with some text that - now we are using V3. If you were using V2, here is the upgrade guide...

File: src/elements/README.md [hunk 4]
```diff
@@ -7,12 +7,3 @@ The UI Elements use [react-intl](https://github.com/yahoo/react-intl) for intern
...
-```js static
-import { addLocaleData } from 'react-intl';
-import frLocaleData from 'react-intl/locale-data/fr';
-addLocaleData(frLocaleData);
-```
-
-The messages that are needed for the components can be imported from `box-ui-elements/i18n/[LANGUAGE]` package under `node_modules`. The code examples for each of the UI Elements assume `en-US` and show how the US english messages are imported in.
```
Comment (priyajeet) [inline src/elements/README.md]: This text will still be needed

**Comment by laughnan** [comment]
closing this PR in favor of https://github.com/box/box-ui-elements/pull/1899

**Pull request closed without merging**